noir-projects/aztec-nr/aztec/src/encrypted_logs/log_assembly_strategies/default_aes128/note/note.nr

```diff
@@ -0,0 +1,222 @@
+ *
+ *             ciphertext: aes_encrypt(
+ *                 final_pt: [
+ *                     pt: {
+ *                         note_bytes: {
+ *                             storage_slot: [u8; 32],
+ *                             note_type_id: [u8; 32],
+ *                             ...note: [u8; N * 32],
+ *                         }: [u8; N * 32 + 64],
+ *                     }: [u8; N * 32 + 64],
+
+ *                 ]: [u8; N * 32 + 64],
+ *
+ *                 // the aes128_encrypt fn automatically inserts padding:
+ *                 pt_aes_padding: [u8; 16 - ( (|pt_length| + |pt|) % 16 )]
+ *
+ *             ): [u8; |pt| + |pt_aes_padding|]
+ *                [u8; |ct|]
+ *
+ *             log_bytes_padding_to_mult_31: [u8; 31 * ceil((1 + 48 + |ct|)/31) - (1 + 48 + |ct|)],
+ *                                           [u8; p]
+ *
+ *         ]: [u8; 1 + 48 + |ct| + p]
+ *
+ *     ): [Field; (1 + 48 + |ct| + p) / 31]
+ *
+ *     log_fields_padding: [Field; PRIVATE_LOG_SIZE_IN_FIELDS - 2 - (1 + 48 + |ct| + p) / 31],
+ *
+ * ]: [Field; PRIVATE_LOG_SIZE_IN_FIELDS]
+ *
+ *
+ */
+
+/********************************************************/
+// End of disgusting arithmetic on generics
+/********************************************************/
+
+// TODO: it feels like this existence check is in the wrong place. In fact, why is it needed at all? Under what circumstances have we found a non-existent note being emitted accidentally?
+fn assert_note_exists<let N: u32>(context: PrivateContext, note_hash_counter: u32) {
+    // TODO(#8589): use typesystem to skip this check when not needed
+    let note_exists =
+        context.note_hashes.storage().any(|n: NoteHash| n.counter == note_hash_counter);
+    assert(note_exists, "Can only emit a note log for an existing note.");
+}
+
+/// This particular log assembly strategy (AES 128) requires the note (and the
+/// note_id and the storage_slot) to be converted into bytes, because the aes function
+/// operates on bytes; not fields.
+/// NB: The "2" in "N + 2" is for the note_id and the storage_slot of the note:
+fn compute_note_plaintext_for_this_strategy<Note, let N: u32>(
+    note: Note,
+    storage_slot: Field,
+) -> [Field; (N + 2)]
+where
+    Note: NoteType + Packable<N>,
+{
+    let packed_note = note.pack();
+
+    let mut fields = [0; N + 2];
+    fields[0] = storage_slot;
+    fields[1] = Note::get_id(); // TODO(#10952): The note type ID can be reduced to 7 bits
+    for i in 0..packed_note.len() {
+        fields[i + 2] = packed_note[i];
+    }
+
+    fields
+}
+
+pub fn compute_log<Note, let N: u32>(
+    context: PrivateContext,
+    note: Note,
+    storage_slot: Field,
+    recipient: AztecAddress,
+    sender: AztecAddress,
+) -> [Field; PRIVATE_LOG_SIZE_IN_FIELDS]
+where
+    Note: NoteType + Packable<N>,
+{
+    let final_plaintext = compute_note_plaintext_for_this_strategy(note, storage_slot);
+
+    encrypt_log(context.this_address(), final_plaintext, recipient, sender)
```

Comment:
Why was this function extracted into its own file?
One of my goals when I refactored encryption was for everything to be in one long, flattened file, so that it was easier to see the whole unravelled log assembly process, whilst we did all our other refactoring. Please can we revert this particular change?
The `encrypt_log` function that's been extracted has a name which isn't descriptive of what the function does: it computes a log (so should live in this `compute_log` function) (Or actually, it assembles a log, so maybe `compute_log` should be renamed to `assemble_log`). A "log" can't be encrypted; a plaintext can be encrypted. And the extracted `encrypt_log` function also computes a header and encrypts that and then concatenates data according to this log assembly strategy.

Reply:
I needed to unit test it against the decryption functionality and during decryption we don't spit out a note but a log plaintext (see process_log function). This is not decided yet but the `process_log` contract function will most likely process event logs as well and there we will need 1 decryption functionality shared between events and notes.

Agree that the name could be improved.